src/Components/Components/src/ComponentFactory.cs

```diff
@@ -5,6 +5,7 @@
 using System.Diagnostics.CodeAnalysis;
```

Comment:
Conceptually, the changes in this file are:

 * Extend the existing per-component-type cache to also track the render mode declared on the component type, so there's no new runtime cost for determining this (not even an extra cache lookup)
 * When a nonnull rendermode is found, use the new `RenderModeResolver` DI service to do the component instantiation. This is how a rendermode gets translated into concrete, platform-specific behavior.